docs/cloud_build_standards.md

```diff
@@ -0,0 +1,94 @@
+| `TYPE` | Service type | CI | No |
+| `ENV` | Environment | dev, prod, uat, ppd | No |
+| `CLUSTER` | Cluster Name | cluster1, cluster2 | Yes |
+| `NODE` | Node within the cluster | node1 | Yes |
+| `APPSTAGE` | Stage of the application and resources associated | Alpha, Beta, Live | Yes |
+| `ACCESS` | Private or Public | private, public | no |
+| `PROJECT-SERVICE` | name of the service | projectx, projecty | No |
+| `PROJECT-PORTFOLIO` | Name of the portfolio | portfoliox | yes |
+| `SERVICE-TYPE` | Type of service | db, webapp, proxy | No |
+| `AZ` | availability zone number | 1a, 1b, 1c | No |
+| `COST-CODE` | cost code if a shared account | 1234-5678 | yes |
+| `REPO` | Repository related to the creation | repo1 | yes |
+| `TEAM` | Name of the team if a shared account | team1 | yes |
+
+### Additional Tagging Naming Standards
+
+| Type | Description | Usage | Optional |
+| ---- | ------ | ------ | ------|
+| `RDS-TYPE` | database type | Postgres, mysql, oracle | RDSTYPE: Oracle | No |
+
+### Automation and Cloud Rules
+
+- There **MUST NOT** be manual changes to any resources everything should be done via code
+- Console access **MUST** be locked down to readonly to stop manual changes or changes being made that are not tracked via code
+- All code **SHOULD** be triggered via a CI job and not run manually
+- All code **MUST** be tested, linted, and planned
+- Terraform **SHOULD** be used to manage cloud infrastructure
+- Baking images into AMI's using packer **SHOULD** be used over running configuration management tools post infrastructure build phase [see here for sharing AMI's between accounts](https://github.com/chrisns/packer-encrypt-copy)
+- Cloud config is preferred where CoreOS is used to bootstrap the OS
+- Cloud-init **SHOULD** be used in exchange for baking images as AMI's, though building AMI's **SHOULD** be the preferred choice
+- Docker / Containerisation **MUST** be used where possible over native EC2
+- **MUST** use multiple [account security structures](https://aws.amazon.com/answers/account-management/aws-multi-account-security-strategy/)
+- All accounts **MUST** use the central IAM structure and policies [please read the varied docs on cross account access](https://aws.amazon.com/blogs/security/tag/cross-account-access/)
+- All user accounts **MUST** have MFA associated with them
+- All user acounts **MUST** assume role and use short lived access keys into the account
+- All CI robot accounts **MUST** have locked down policies to restrict to IP's of CI plus detailed policies where possible and multiple robot accounts to reduce risk where possible
+- All accounts **MUST** log to central audit
+- Firewall management **SHOULD** be as close to the application as possible and not within generic cloud infrastructure (it becomes difficult to manage when there are multiple services sharing)
+- All services **MUST** traverse a AWS test account before moving into the AWS production account with validated tests
```

Comment:
Maybe something more explicit to say that accounts must be provisioned centrally, you can't go and procure some amazon resource through some other channel, it all has to go via IPT/ACP?